src/apphosting/config.ts

```diff
@@ -33,15 +38,15 @@ export interface Config {
 }
 
 /**
- * Finds the path of apphosting.yaml.
- * Starts with cwd and walks up the tree until apphosting.yaml is found or
- * we find the project root (where firebase.json is) or the filesystem root;
- * in these cases, returns null.
+ * Finds the project root for apphosting backends.
+ * Starts with cwd and walks up the path until a apphosting.yaml file is found
+ *
+ * Eample path that's returned: "/home/my-project"
  */
-export function yamlPath(cwd: string): string | null {
+export function discoverProjectRoot(cwd: string): string | null {
```

Comment:
projectRoot or backendRoot? is firebase.json guaranteed to be colocated with apphosting.yaml?

Reply:
yes backend root makes more sense. I don't think firebase.json needs to be colocated wiht apphosting.yaml. We can assume the backend root is wherever apphosting.yaml is.